WalletWasabi.Fluent/Behaviors/ShowAttachedFlyoutWhenFocusedBehavior.cs

```diff
@@ -0,0 +1,133 @@
+using System.Linq;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using Avalonia;
+using Avalonia.Controls;
+using Avalonia.Controls.Diagnostics;
+using Avalonia.Controls.Primitives;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using Avalonia.VisualTree;
+using ReactiveUI;
+using WalletWasabi.Fluent.Extensions;
+using WalletWasabi.Fluent.Helpers;
+
+namespace WalletWasabi.Fluent.Behaviors;
+
+public class ShowAttachedFlyoutWhenFocusedBehavior : AttachedToVisualTreeBehavior<Control>
+{
+	public static readonly DirectProperty<ShowAttachedFlyoutWhenFocusedBehavior, bool> IsFlyoutOpenProperty =
+		AvaloniaProperty.RegisterDirect<ShowAttachedFlyoutWhenFocusedBehavior, bool>(
+			"IsFlyoutOpen",
+			o => o.IsFlyoutOpen,
+			(o, v) => o.IsFlyoutOpen = v);
+
+	private bool _isFlyoutOpen;
+
+	public bool IsFlyoutOpen
+	{
+		get => _isFlyoutOpen;
+		set => SetAndRaise(IsFlyoutOpenProperty, ref _isFlyoutOpen, value);
+	}
+
+	protected override void OnAttachedToVisualTree(CompositeDisposable disposable)
+	{
+		if (AssociatedObject is null)
+		{
+			return;
+		}
+
+		if (AssociatedObject.GetVisualRoot() is not Control visualRoot)
+		{
+			return;
+		}
+
+		var flyoutBase = FlyoutBase.GetAttachedFlyout(AssociatedObject);
+		if (flyoutBase is null)
+		{
+			return;
+		}
+
+		var controller = new FlyoutShowController(AssociatedObject, flyoutBase).DisposeWith(disposable);
+
+		FocusBasedFlyoutOpener(AssociatedObject, flyoutBase, controller).DisposeWith(disposable);
+		IsOpenPropertySynchronizer(controller).DisposeWith(disposable);
+
+		// EDGE CASES
+		// Edge case when the Visual Root becomes active and the Associated object is focused.
+		ActivateOpener(AssociatedObject, visualRoot, controller).DisposeWith(disposable);
+		DeactivateCloser(visualRoot, controller).DisposeWith(disposable);
+
+		// This is a workaround for the case when the user switches theme. The same behavior is detached and re-attached on theme changes.
+		// If you don't close it, the Flyout will show in an incorrect position. Maybe bug in Avalonia?
+		if (IsFlyoutOpen)
+		{
+			controller.SetIsOpen(false);
+		}
+	}
+
+	private static IDisposable DeactivateCloser(Control visualRoot, FlyoutShowController controller)
+	{
+		return Observable.FromEventPattern(visualRoot, nameof(Window.Deactivated))
+			.Do(_ => controller.SetIsOpen(false))
+			.Subscribe();
+	}
+
+	private static IDisposable ActivateOpener(
+		IInputElement associatedObject,
+		Control visualRoot,
+		FlyoutShowController controller)
+	{
+		return Observable.FromEventPattern(visualRoot, nameof(Window.Activated))
+			.Do(
+				_ =>
+				{
+					if (associatedObject.IsFocused)
+					{
+						controller.SetIsOpen(true);
+					}
+				})
+			.Subscribe();
+	}
```

Comment:
You can ignore it, for me, it feels easier to read.

```suggestion
	private static IDisposable ActivateOpener(IInputElement associatedObject, Control visualRoot, FlyoutShowController controller)
	{
		return Observable.FromEventPattern(visualRoot, nameof(Window.Activated))
			.Where(_ => associatedObject.IsFocused)
			.Do(_ => controller.SetIsOpen(true))
			.Subscribe();
	}
```

Reply:
Wonderful! I also prefer that syntax. It's way more readable. I don't know why I didn't refactor this before. Thankfully we have such good reviews :D